src/Aspire.Hosting.Azure/ExistingAzureResourceAnnotation.cs

```diff
@@ -0,0 +1,18 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+
+namespace Aspire.Hosting.Azure;
+
+/// <summary>
+/// Represents a resource that is not managed by Aspire's provisioning or
+/// container management layer.
+/// </summary>
+public class ExistingAzureResourceAnnotation(IResourceBuilder<ParameterResource> nameParameter) : IResourceAnnotation
+{
+    /// <summary>
+    /// Gets the name of the existing resource.
+    /// </summary>
+    public IResourceBuilder<ParameterResource> NameParameter { get; } = nameParameter;
+}
```

Comment:
```suggestion
public class ExistingAzureResourceAnnotation(ParameterResource nameParameter) : IResourceAnnotation
{
    /// <summary>
    /// Gets the name of the existing resource.
    /// </summary>
    public ParameterResource NameParameter { get; } = nameParameter;
}
```

Reply:
We don't have an `AsProvisioningParameter` overload that takes the `ParameterResource` directly although that seems like an oversight given what the implementation currently does.

https://github.com/dotnet/aspire/blob/5008bfb621ee0283fdbe5c144a2d58341ff01e54/src/Aspire.Hosting.Azure/AzureProvisioningResourceExtensions.cs#L67-L78

I think we can probably just add another overload that takes `ParameterResource` and have the existing one call into it.